--- a/web/viewer.js
+++ b/web/viewer.js
@@ -695,7 +695,8 @@
     kind: OptionKind.VIEWER + OptionKind.PREFERENCE
   },
   enableScripting: {
-    value: true,
+    // Odoo: don't support scripting (#115302)
+    value: false,
     kind: OptionKind.VIEWER + OptionKind.PREFERENCE
   },
   enableSignatureEditor: {
@@ -715,7 +716,8 @@
     kind: OptionKind.VIEWER
   },
   externalLinkTarget: {
-    value: 0,
+    // Odoo: open links in new tabs to keep odoo document (#84594)
+    value: 2,
     kind: OptionKind.VIEWER + OptionKind.PREFERENCE
   },
   highlightEditorColors: {
@@ -791,7 +793,8 @@
     kind: OptionKind.VIEWER + OptionKind.PREFERENCE
   },
   viewerCssTheme: {
-    value: 0,
+    // Odoo
+    value: document.cookie.includes("color_scheme=dark") ? 2 : 1,
     kind: OptionKind.VIEWER + OptionKind.PREFERENCE
   },
   viewOnLoad: {
@@ -2574,7 +2577,7 @@
   overlayAttributes(fromElement, toElement);
   return toElement;
 }
-;// ./node_modules/cached-iterable/src/cached_iterable.js
+;// ./node_modules/cached-iterable/src/cached_iterable.mjs
 class CachedIterable extends Array {
   static from(iterable) {
     if (iterable instanceof this) {
@@ -2583,7 +2586,7 @@
     return new this(iterable);
   }
 }
-;// ./node_modules/cached-iterable/src/cached_sync_iterable.js
+;// ./node_modules/cached-iterable/src/cached_sync_iterable.mjs
 
 class CachedSyncIterable extends CachedIterable {
   constructor(iterable) {
@@ -2618,7 +2621,7 @@
     return this[this.length - 1];
   }
 }
-;// ./node_modules/cached-iterable/src/cached_async_iterable.js
+;// ./node_modules/cached-iterable/src/cached_async_iterable.mjs
 
 class CachedAsyncIterable extends CachedIterable {
   constructor(iterable) {
@@ -2655,7 +2658,7 @@
     return this[this.length - 1];
   }
 }
-;// ./node_modules/cached-iterable/src/index.js
+;// ./node_modules/cached-iterable/src/index.mjs
 
 
 ;// ./node_modules/@fluent/dom/esm/localization.js
@@ -7515,6 +7518,8 @@
     this.#mainContainer = mainContainer;
     const checkedInputs = new Map([[this.highlightAll, "highlightallchange"], [this.caseSensitive, "casesensitivitychange"], [this.entireWord, "entirewordchange"], [this.matchDiacritics, "diacriticmatchingchange"]]);
     this.toggleButton.addEventListener("click", () => {
+    // ODOO PATCH PRINT PREVIEW MOBILE
+    this.hasFinishPrint = null;
       this.toggle();
     });
     this.findField.addEventListener("input", () => {
@@ -9120,16 +9125,24 @@
   }
   performPrint() {
     this.throwIfInactive();
-    return new Promise(resolve => {
-      setTimeout(() => {
-        if (!this.active) {
-          resolve();
-          return;
-        }
-        print.call(window);
+    // ODOO PATCH PRINT PREVIEW MOBILE
+    const hasFinishPrintPromise = new Promise((resolve) => {
+      if ("afterprint" in window) {
+        this.hasFinishPrint = resolve;
+      } else {
         setTimeout(resolve, 20);
-      }, 0);
+      }
     });
+    setTimeout(() => {
+      if (!this.active) {
+        // ODOO PATCH PRINT PREVIEW MOBILE
+        this.hasFinishPrint();
+        return;
+      }
+      print.call(window);
+    }, 0);
+    // ODOO PATCH PRINT PREVIEW MOBILE
+    return hasFinishPrintPromise;
   }
   get active() {
     return this === activeService;
@@ -9161,7 +9174,15 @@
       });
     } else {
       const activeServiceOnEntry = activeService;
-      activeService.renderPages().then(() => activeServiceOnEntry.performPrint()).catch(() => {}).then(() => {
+      // ODOO: FIX MOBILE PRINT PREVIEW
+      const timeBeforeRendering = new Date().getTime();
+      activeService.renderPages().then(() => 
+        // ODOO: FIX MOBILE PRINT PREVIEW
+        Promise.all([
+          activeServiceOnEntry.performPrint(),
+          new Promise(resolve => setTimeout(resolve, 1000 + new Date().getTime() - timeBeforeRendering))
+        ])
+      ).catch(() => {}).then(() => {
         if (activeServiceOnEntry.active) {
           abort();
         }
@@ -9202,6 +9223,11 @@
 }, true);
 if ("onbeforeprint" in window) {
   const stopPropagationIfNeeded = function (event) {
+    // ODOO PATCH PRINT PREVIEW MOBILE
+    if (activeService?.hasFinishPrint && event.type === "afterprint") {
+      activeService.hasFinishPrint();
+      return;
+    }
     if (event.detail !== "custom") {
       event.stopImmediatePropagation();
     }
@@ -19256,7 +19282,8 @@
     });
     pagesPromise.then(() => {
       this._unblockDocumentLoadEvent();
-      this._initializeAutoPrint(pdfDocument, openActionPromise);
+      // Odoo: don't support scripting (#115302)
+      // this._initializeAutoPrint(pdfDocument, openActionPromise);
     }, reason => {
       this._documentError("pdfjs-loading-error", {
         message: reason.message
